team-manager/src/test/unit/java/ua/knu/knudev/teammanager/service/DepartmentServiceTest.java

```diff
@@ -151,4 +157,163 @@ void should_throwDepartmentException_When_DepartmentDoesNotExistOnGet() {
 
         verify(departmentRepository, times(1)).findById(TEST_DEPARTMENT_ID);
     }
+
+    @Test
+    @DisplayName("Should create department when given valid data")
+    public void should_createDepartment_When_givenValidData() {
+        SpecialtyCreationDto specialtyCreationDto1 = new SpecialtyCreationDto(14.378, "Computer Engineering");
+        SpecialtyCreationDto specialtyCreationDto2 = new SpecialtyCreationDto(14.378, "Radio technics");
+        SpecialtyCreationDto specialtyCreationDto3 = new SpecialtyCreationDto(14.378, "History");
+
+        DepartmentCreationRequest departmentCreationRequest = buildDepartmentCreationRequest(
+                Set.of(specialtyCreationDto1, specialtyCreationDto2, specialtyCreationDto3),
+                TEST_DEPARTMENT_NAME
+        );
+
+        departmentService.createDepartment(departmentCreationRequest);
+
+        ArgumentCaptor<Department> departmentCaptor = ArgumentCaptor.forClass(Department.class);
+        verify(departmentRepository, times(1)).save(departmentCaptor.capture());
+
+        Department savedDepartment = departmentCaptor.getValue();
+
+        assertEquals(TEST_DEPARTMENT_NAME, savedDepartment.getName());
+        assertEquals(3, savedDepartment.getSpecialties().size());
+
+        assertTrue(savedDepartment.getSpecialties().stream()
+                .anyMatch(specialty -> "Computer Engineering".equals(specialty.getName())));
+        assertTrue(savedDepartment.getSpecialties().stream()
+                .anyMatch(specialty -> "Radio technics".equals(specialty.getName())));
+        assertTrue(savedDepartment.getSpecialties().stream()
+                .anyMatch(specialty -> "History".equals(specialty.getName())));
+    }
+
+
+    @Test
+    @DisplayName("Should not create department when not given department name")
+    public void should_notCreateDepartment_When_notGivenDepartmentName() {
+        SpecialtyCreationDto specialtyCreationDto1 = new SpecialtyCreationDto(14.378, "Computer Engineering");
+        SpecialtyCreationDto specialtyCreationDto2 = new SpecialtyCreationDto(14.378, "Radio technics");
+        SpecialtyCreationDto specialtyCreationDto3 = new SpecialtyCreationDto(14.378, "History");
+
+        DepartmentCreationRequest departmentCreationRequest = buildDepartmentCreationRequest(
+                Set.of(specialtyCreationDto1, specialtyCreationDto2, specialtyCreationDto3),
+                null
+        );
+
+        ConstraintViolationException constraintViolationException = assertThrows(
+                ConstraintViolationException.class, () -> {
+                    departmentService.createDepartment(departmentCreationRequest);
+                });
+
+        assertEquals("Department name cannot be blank or empty", constraintViolationException.getMessage());
+    }
+
+    @Test
+    @DisplayName("Should not create department when department name is blank")
+    public void should_notCreateDepartment_When_departmentNameIsBlank() {
```

Comment:
Remove all the validation tests. And add 1 more test when department already exists

Reply:
done